digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala

```diff
@@ -367,6 +367,39 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: VVHClient,
     }
   }
 
+  private val getRoadsForRoadAddressBrowser: SwaggerSupportSyntax.OperationBuilder = (
+    apiOperation[Map[String,Any]]("getRoadsForRoadAddressBrowser").parameters(
+      queryParam[String]("startDate").description("Situation date (dd-MM-yyyy)"),
+      queryParam[Long]("ely").description("Ely number of a road address").optional,
+      queryParam[Long]("roadNumber").description("Road Number of a road address").optional,
+      queryParam[Long]("minRoadPartNumber").description("Road Part Number of a road address").optional,
+      queryParam[Long]("maxRoadPartNumber").description("Road Part Number of a road address").optional
+    )
+      tags "ViiteAPI - Road Address Browser"
+      summary "Returns all roads that fill the search criteria"
+  )
+
+  get("/roadaddressbrowser/roads", operation(getRoadsForRoadAddressBrowser)) {
+    try {
+      val startDate = params.get("startDate").get
+      val ely = params.get("ely").map(_.toLong)
+      val roadNumber = params.get("roadNumber").map(_.toLong)
+      val minRoadPartNumber = params.get("minRoadPartNumber").map(_.toLong)
+      val maxRoadPartNumber = params.get("maxRoadPartNumber").map(_.toLong)
+
```

Comment:
Tähän väliin karkeat parametritarkistukset.

startDaten ja elyn tarkastamisesta en tosin tiedä - voisi niillekin toki jotkut karkeat arviot heittää. (Onko enumeraatiolistaan kuuluminen tarkistettavissa helposti? ja päiväystä voisi rajata vaikka karkeasti 1900-luvulta eteenpäin, ja tietty formaattitarkistus.)
Tienumeron tulee olla 1-99999-välillä,
Tieosanumerot 1-999, ja min ja max -tieosanumerot pitää olla min<max,

Jos kriteerit feilaavat, paluuarvona joku 400 Bad request-virhe.
Nää voi tosin heittää omaan puolariinsa tai vaikka omaan tikettiinsäkin.

Reply:
Kyl varmaan kuuluu tähän. Vähän periaateella tehdään aluksi hyvin.

Reply:
Lisä ominaisuudet sitten muuhun pull request.

Reply:
Sinänsä näiden arvojen "oikeellisuuden" tarkastaminen on turhaa sillä kantaan ei kirjoiteta mitään, sieltä vain haetaan ja jos käyttäjä päättää hakea 99999999 arvolla tietä niin tulokseksi palautuu tyhjä lista. 

Käyttöliittymäpuolella on input fieldit ovat type=number joten niihin ei käy mikään muu kuin numero
startDate on myös required kenttä käyttöliittymäpuolella joten ilman sitä haku ei päädy edes backendille asti.

Ely tai Tienumero kenttä tulee myös olla täytettynä ja tämäkin validoidaan käyttöliittymän puolella ennen kuin tehdään kutsu endpointtiin. 

Lisäsin käyttöliittymään elylle, roadNumberille ja roadPartNumbereille nuo substanssi validoinnit

Reply:
(Tästä keskusteltiin: Validointien puuttuminen ei ole show stopper, mutta olisi hyvien tapojen mukaista. 

tarkistuksia vastaan:
-lmankin toimii, tsekit on jo UI:n puolella, käyttäjät ovat ammattikäyttäjiä, eli tietänevät hakuhaarukat kyllä, ja virheet näkee lokeista jos niitä tulee

tarkistuksien puolesta:
-Parametrien oikeellisuuden olettaminen on rumaa, bäkkärin sisääntulokerros on oikea paikka niiden tsekkaamiseen. Ammattikäyttäjillekin voi tarjota hyvän käyttöliittymän, typoja tekevät hekin. Debuggaus on helpompaa jos virhe palautuu heti siellä missä vika on - tai siltä vältytään kokonaan, jos virhe saadaan ulos heti.
Yleistasolla käyttöliittymän irrottaminen toiminnallisuudesta kuuluu best practiceihin (model-view-controller, tai ihan vaan UI/Backend) - käyttöliitymä pitäisi voida vaihtaa toiseen bäkkärin toiminnallisuuden rapautumatta.
)